src/transformers/models/bart/modeling_tf_bart.py

```diff
@@ -922,7 +940,14 @@ def call(
         past_key_values_length = shape_list(past_key_values[0][0])[2] if past_key_values is not None else 0
 
         # embed positions
-        positions = self.embed_positions(input_shape, past_key_values_length)
+        if position_ids is None:
+            if past_key_values is not None:
+                raise ValueError("Make sure to provide `decoder_position_ids` when passing `past_key_values`.")
+            positions = self.embed_positions(input_shape, past_key_values_length)
+        else:
+            positions = self.embed_positions(input_shape, position_ids=position_ids)
```

Comment:
~~why the `else` branch don't need to pass `past_key_values_length`~~

OK, I got it. If `past_key_values_length` is passed `TFBartDecoder`, we need `position_ids`, and no more need `past_key_values_length` for `self.embed_positions`